File: server/match_history.js
import axios from 'axios';
import send_server_error from './network.js'

export default async function match_history(req, res) {
    const { name, tag, region } = req.query;

    try {
        const user = await axios.get(`https://${region}.api.riotgames.com/riot/account/v1/accounts/by-riot-id/${name}/${tag}?api_key=${process.env.RIOT_KEY}`);
        const { puuid } = user.data;
        const history = await axios.get(`https://${region}.api.riotgames.com/lol/match/v5/matches/by-puuid/${puuid}/ids?queue=420&start=0&count=20&api_key=${process.env.RIOT_KEY}`);
        const match_ids = history.data;

        if (match_ids.length == 0) {
            return res.status(403).json({ error: 'No games found'});
        }

        res.json({ puuid, match_ids });
    } catch (error) {
        if (error.response) {
            if (error.response.status === 404) {
                res.status(404).json({ error: 'User not found' });
            } else {
                console.error('API error:', error.response.status, error.response.data);
                send_server_error(res);
            }
        } else {
            console.error('Network error:', error.message);
            send_server_error(res);
        }
    }
}

File: server/match_history.py
import os, requests
from flask import jsonify, request
from server.network import send_server_error

def match_history():
    name = request.args.get('name')
    tag = request.args.get('tag')
    region = request.args.get('region')
    queue = request.args.get('queue')

    queue_id = {
        'soloq': 420,
        'draft': 400,
        'flex': 440,
    }.get(queue)

    try:
        user_response = requests.get(
            f'https://europe.api.riotgames.com/riot/account/v1/accounts/by-riot-id/{name}/{tag}',
            params={'api_key': os.environ.get('RIOT_KEY')}
        )

        if user_response.status_code == 404:
            return jsonify({'error': 'User not found'}), 404

        user_response.raise_for_status()
        puuid = user_response.json()['puuid']

        history_response = requests.get(
            f'https://{region}.api.riotgames.com/lol/match/v5/matches/by-puuid/{puuid}/ids',
            params={
                'queue': queue_id,
                'start': 0,
                'count': 20,
                'api_key': os.environ.get('RIOT_KEY')
            }
        )

        history_response.raise_for_status()
        match_ids = history_response.json()

        if len(match_ids) == 0:
            return jsonify({'error': 'No games found'}), 403

        return jsonify({
            'puuid': puuid,
            'match_ids': match_ids
        })

    except requests.exceptions.RequestException as error:
        print('API / Network error:', str(error))
        return send_server_error()
